plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -97,17 +106,30 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
         deferredResults.any { it.await() }
     }
 
+    private suspend fun ignoreFileByExtension(file: File, scope: CoroutineScope): Boolean = with(scope) {
+        return !isFileExtensionAllowed(file)
+    }
+
+    suspend fun ignoreFileByExtension(file: VirtualFile, scope: CoroutineScope): Boolean = ignoreFileByExtension(File(file.path), scope)
```

Comment:
if `VirtualFile#getExtension` exists, some of this seems unnecessary

Reply:
agreed, will refactor